common/djangoapps/third_party_auth/utils.py

```diff
@@ -0,0 +1,65 @@
+from random import randint
+
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+
+from django.contrib.auth.models import User
+
+
+class UsernameGenerator(object):
+    """Allows customize the way that the username is created based on the fullname"""
+
+    def __init__(self):
+        self.separator_character = '_'
+        self.in_lowercase = True
+        self.random = True
+
+    def insert_separator(self, fullname):
+        """
+        Allows set a custom separator character for each whitespac
+        in the fullname string. By default, all whitespaces are removed.
+        """
+        username = fullname.replace(' ', self.separator_character)
+        return username
+
+    def lower(self, username):
```

Comment:
Since this method could not return a lower case string, we should change its name. Maybe something like `process_username`